lua/nvim-treesitter/textobjects.lua

```diff
@@ -66,13 +66,41 @@ function M.select_textobject(query_string)
     if smallest_range.start then
       local start_range = {smallest_range.start.node:range()}
       local node_range = {smallest_range.node:range()}
-      ts_utils.update_selection(bufnr, {start_range[1], start_range[2], node_range[3], node_range[4]})
+      return bufnr, {start_range[1], start_range[2], node_range[3], node_range[4]}, smallest_range.node
     else
-      ts_utils.update_selection(bufnr, smallest_range.node)
+      return bufnr, {smallest_range.node:range()}, smallest_range.node
     end
   end
 end
 
+function M.select_textobject(query_string)
+  local bufnr, textobject = get_textobject_at_point(query_string)
+  if textobject then
+    ts_utils.update_selection(bufnr, textobject)
+  end
+end
+
+function M.swap_textobject(query_string, direction)
+  local bufnr, textobject_range, node = get_textobject_at_point(query_string)
+  local step = (direction > 0 and 1 or -1)
```

Comment:
not sure you need the braces here

Reply:
I guess the linter was happier with it, but will investiage.